### ISSUE TITLE
New False Negative for DM_STRING_CTOR

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#1766]
I found a false negative for this rule. I think we should consider this similar API because it is the same with new String() and new String(java.lang.String). This program can lead to lower performance.

The rule link: [link](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-method-invokes-inefficient-new-string-string-constructor-dm-string-ctor). Please refer to the following code sample:

```java
public void test(StringBuilder builder) {
    String str = new String(builder);  // Need report warning
    System.out.println(str);
}
```

My SpotBugs version is 4.4.2 and I think it needs report a warning for the line 3, but no expected outcome. Thanks for your consideration.

**Comment by gtoison** [comment]
The code for `StringBuilder.toStrin()` is:
```
    public String toString() {
        // Create a copy, don't share the array
        return new String(this);
    }
```

So I agree it is not a false negative: calling `new String(builder)` is the same
